frappe/core/doctype/file/file.py

```diff
@@ -351,3 +352,21 @@ def check_file_permission(file_url):
 
 	raise frappe.PermissionError
 
+@frappe.whitelist()
+def extraxt_uploaded_file(file_name, ):
+	if ".zip" in file_name:
+		zip_path = frappe.get_site_path("public", "files")
+		zip_path_with_filename=frappe.get_site_path("public", "files", file_name)
+		with zipfile.ZipFile(zip_path_with_filename) as zf:
+			zf.extractall(zip_path)
+			for info in zf.infolist():
+				data = info.filename
+				file_doc=frappe.new_doc("File")
+				file_doc.file_name=info.filename
+				file_doc.file_size=info.file_size
+				file_doc.folder="Home/Attachments"
```

Comment:
@rmehta I have checked this, but unable to get current folder name in file list.
cur_frm.doc.folder returns null on file list.